raidsplit/bot_commands.py

```diff
@@ -0,0 +1,172 @@
+                admin=UserId(ctx.author.id),
+                proposals={},
+                member_infos={},
+                is_open=True)
+
+    await ctx.send(f"A raid wit the name **{raid_name}** and **{len(raiders)}** members was created.")
+
+    for raider in raiders:
+        new_member: UserId = UserId(raider.id)
+        raid.add_member(raider)
+        await raider.send(
+            f"Hi {raid.member_infos[new_member].name}. Your input has been requested for the **{raid_name}** raid.\n"
+            "Please use the `!split allocate <raid_name> <member_handle> <allocation>`. "
+            "Your allocations should be a number greater 0 and smaller 100 \n\n"
+            "If you make a mistake use the `!split allocate` again to modify your allocation."
+        )
+
+    LocalFile().store(raid)
+
+
+@split.command(help="Show all raid members.")
+async def members(ctx, raid_name):
+    """
+    This command returns all members. It is intended to be used before submitting the allocations.
+    """
+    raid = LocalFile().find_raid(raid_name)
+    member_names = raid.get_member_names()
+
+    await ctx.send(f"The raid **{raid_name}** has **{len(member_names)}** members:\n"
+                   f"*{', '.join(member_names)}*\n"
+                   f"Use the command `allocate` to specify their shares in the same order.")
+
+
+@split.command(
+    help="Allocate percentages to all raid members as ordered in the `!split members <raid_name>` command.")
+async def allocate(ctx, raid_name, percentages: commands.Greedy[float]):
+    """
+    This command allocates shares to all members specified in the order of the members command.
+    """
+    raid = LocalFile().find_raid(raid_name)
+
+    if not raid.is_open:
+        return await ctx.send(f"The raid **{raid_name}** has already been closed.")
+
+    members_ids = raid.get_member_ids()
+
+    if len(members_ids) is not len(percentages):
+        await ctx.send(f"The number of allocations **{len(percentages)}** "
+                       f"does not match the number of **{len(members_ids)}**.")
+        await members(ctx, raid_name)
+        return
+    elif sum(percentages) != 100.0:
+        await ctx.send(f"The percentages must add up to **{100}**.")
+        return
+    elif any(x <= 0 or x >= 100 for x in percentages):
+        await ctx.send(f"The percentages must greater than 0 and smaller than 100.")
+        return
+
+    sender: UserId = UserId(ctx.author.id)
+    # iterate over senders beneficiaries in member info
+    for i, beneficiary in enumerate(members_ids):
+        raid.proposals[sender][beneficiary] = percentages[i]
+
+    raid.update_shares()
+    LocalFile().store(raid)
+
+    await show(ctx, raid_name)
+
+
+@split.command(help="Show your averaged shares.")
+async def share(ctx, raid_name):
+    raid = LocalFile().find_raid(raid_name)
+    sender = UserId(ctx.author.id)
+
+    all_calculated = raid.calculate_share(sender)
+
+    if not all_calculated:
```

Comment:
It may make sense to calculate even if some allocations are missing, but to also send a warning up with it